dashboard/api_edx_cache.py

```diff
@@ -342,7 +342,7 @@ def update_all_cached_grade_data(cls, user):
             None
         """
         # get the credentials for the current user for edX
-        user_social = user.social_auth.get(provider=EdxOrgOAuth2.name)
+        user_social = user.social_auth.filter(provider=EdxOrgOAuth2.name).latest('id')
```

Comment:
This ORM query is repeated in many places (and, of course, it was repeated before you made this fix). Ideally we would be able add a method to the model itself to run this query, but since `User` is provided by Django, I think a good alternative option is to add a new API method to `profiles/api.py`. 

```python
def get_social_auth(user):
    return user.social_auth.filter(provider=EdxOrgOAuth2.name).latest('id')
```

You could replace the queries in this PR with calls to that method, and `get_social_username` in the same module could call it directly